go/git/get.go

```diff
@@ -0,0 +1,202 @@
+		if err != nil {
+			return keybase1.Folder{}, err
+		}
+		var folderType keybase1.FolderType
+		if team.IsPublic() {
+			folderType = keybase1.FolderType_PUBLIC
+		} else {
+			folderType = keybase1.FolderType_PRIVATE
+		}
+		return keybase1.Folder{
+			Name:       name,
+			FolderType: folderType,
+			Private:    !team.IsPublic(),
+		}, nil
+	} else {
+		return keybase1.Folder{
+			Name:       team.Name().String(),
+			FolderType: keybase1.FolderType_TEAM,
+			Private:    !team.IsPublic(),
+		}, nil
+	}
+}
+
+func getMetadataInner(ctx context.Context, g *libkb.GlobalContext, folder *keybase1.Folder) ([]keybase1.GitRepoResult, error) {
+	teamer := NewTeamer(g)
+	cryptoer := NewCrypto(g)
+
+	apiArg := libkb.APIArg{
+		Endpoint:    "kbfs/git/team/get",
+		SessionType: libkb.APISessionTypeREQUIRED,
+		NetContext:  ctx,
+		Args:        libkb.HTTPArgs{
+		// a limit parameter exists, default 100, and we don't currently set it
+		},
+	}
+
+	// The team_id parameter is optional. Add it in if the caller supplied it.
+	if folder != nil {
+		if folder.FolderType != keybase1.FolderType_TEAM {
+			return nil, errors.New("git repo folders must be team folders")
+		}
+		teamIDVis, err := teamer.LookupOrCreate(ctx, *folder)
+		if err != nil {
+			return nil, err
+		}
+		apiArg.Args["team_id"] = libkb.S{Val: string(teamIDVis.TeamID)}
+	}
+	var serverResponse ServerResponse
+	err := g.GetAPI().GetDecode(apiArg, &serverResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	// Initializing the results list to non-nil means that we end up seeing
+	// "[]" instead of "null" in the final JSON output on the CLI, which is
+	// preferable.
+	resultList := []keybase1.GitRepoResult{}
+	for _, responseRepo := range serverResponse.Repos {
+		// If the folder was passed in, use it. Otherwise, load the team to
+		// figure it out.
+		var repoFolder keybase1.Folder
+		if folder != nil {
+			repoFolder = *folder
+		} else {
+			repoFolder, err = folderFromTeamID(ctx, g, responseRepo.TeamID)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		teamIDVis := keybase1.TeamIDWithVisibility{
+			TeamID: responseRepo.TeamID,
+		}
+		if repoFolder.Private {
+			teamIDVis.Visibility = keybase1.TLFVisibility_PRIVATE
+		} else {
+			teamIDVis.Visibility = keybase1.TLFVisibility_PUBLIC
+		}
+
+		ciphertext, err := base64.StdEncoding.DecodeString(responseRepo.EncryptedMetadata)
+		if err != nil {
+			return nil, err
+		}
+
+		nonceSlice, err := base64.StdEncoding.DecodeString(responseRepo.Nonce)
+		if err != nil {
+			return nil, err
+		}
+		if len(nonceSlice) != len(keybase1.BoxNonce{}) {
+			return nil, fmt.Errorf("expected a nonce of length %d, found %d", len(keybase1.BoxNonce{}), len(nonceSlice))
+		}
+		var nonce keybase1.BoxNonce
+		copy(nonce[:], nonceSlice)
+
+		encryptedMetadata := keybase1.EncryptedGitMetadata{
+			V:   1, // version 1 hardcoded for now -- if we create a version 2 we'll need to start storing it server-side
```

Comment:
I'd kind of prefer to start doing this now (storing version server side)

Reply:
Sure thing, I'll whip up a small server-side PR to go with.